cmd/net.go

```diff
@@ -66,18 +67,37 @@ func mustGetLocalIP4() (ipList set.StringSet) {
 // getHostIP4 returns IPv4 address of given host.
 func getHostIP4(host string) (ipList set.StringSet, err error) {
 	ipList = set.NewStringSet()
-	ips, err := net.LookupIP(host)
-	if err != nil {
-		return ipList, err
-	}
-
-	for _, ip := range ips {
-		if ip.To4() != nil {
-			ipList.Add(ip.String())
+	//variable to list resolved IP addresses
+	var ips []net.IP
+	// channel to indicate completion of host resolution
+	doneCh := make(chan struct{})
+	// Indicate to our retry routine to exit cleanly, upon this function return.
+	defer close(doneCh)
+	// Wait on the jitter retry loop
+	retryTimeCh := newRetryTimerSimple(doneCh)
+	for {
```

Comment:
Infinitely retrying could be problem in non-container environment.  I think we could do only for kubernetes and docker swam.